src/aggregation/intermediate_agg_result.rs

```diff
@@ -218,7 +221,7 @@ impl IntermediateBucketResult {
                     })
                     .collect();
 
-                std::mem::swap(entries_left, &mut buckets);
+                std::mem::swap(buckets_left, &mut buckets);
```

Comment:
the use of mem::swap is useless here.

Reply:
buckets_left = buckets;

Reply:
entries_left is a &mut binding from a match pattern and buckets is a temporary variable. I tried some things, but couldn't find a solution other than mem::swap

Reply:
`*entries_left = buckets` did not work?